exec/plugin-commands-script-runners/src/run.ts

```diff
@@ -300,6 +304,28 @@ ${renderCommands(rootScripts)}`
   return output
 }
 
+export interface RunScriptOptions {
+  enablePrePostScripts: boolean
+}
+
+export const runScript: (scriptName: string, manifest: ProjectManifest, lifecycleOpts: RunLifecycleHookOptions, runScriptOptions: RunScriptOptions, passedThruArgs: string[]) => Promise<void> = async function (scriptName, manifest, lifecycleOpts, runScriptOptions, passedThruArgs) {
```

Comment:
move all the arguments to the options except `scriptName`

make options the first argument.

Then you'll able to bind the options and only run the new function with script name in the map at line 224

```ts
const _runScript = runScript.bind(null, {
  ....
})
await Promise.all(specifiedScripts.map(script => limitRun(() => runScript(script))))
```

Reply:
I am not sure I understand your point properly, but is this what you mean? [39f3152](https://github.com/pnpm/pnpm/pull/5871/commits/39f3152f3617233c05bfc2ed3e719dd52786fb47)